### ISSUE TITLE
Cell Constructor

### ISSUE THREAD

**Issue by marthie** [dhorions/boxable#69]
Hi All,

I try to get familiar with boxable. I just want to create a `BaseTable` with two columns. Here is my code:

```java
PDDocument doc = new PDDocument();
		
PDPage p1 = new PDPage(PDRectangle.A4);
doc.addPage(p1);
		
float yTableStart = PDRectangle.A4.getHeight() - 75;
float tableWidth = PDRectangle.A4.getWidth() - 150;
BaseTable table = new BaseTable(yTableStart, yTableStart, 75, tableWidth, 75, doc, p1, true, true);

Row<PDPage> hRow = table.createRow(20f);
hRow.createCell(tableWidth / 2, "Test1").setFont(PDType1Font.COURIER);
hRow.createCell(tableWidth / 2, "Test1").setFont(PDType1Font.COURIER);
		
table.addHeaderRow(hRow);
		
table.draw();
		
doc.save(new File("d:/tmp/table.pdf"));
```
As you can see in my code I am using the whole page width for my table (except a margin space). In the table above I like to arrange two cols/cells with the same width (`tableWidth / 2`).

When I execute the code, I get the following output:
```
Exception in thread "main" java.lang.IllegalArgumentException: Cell Width=991.3519 can't be bigger than row width=445.27563
	at be.quodlibet.boxable.Cell.<init>(Cell.java:103)
	at be.quodlibet.boxable.Cell.<init>(Cell.java:68)
	at be.quodlibet.boxable.Row.createCell(Row.java:47)
	at com.commerzbank.tryPdfBox.App4.main(App4.java:30)
```

After some debugging I found out, that the parameterized cell width is increased by the follwoing lines:

```java
Cell(Row<T> row, float width, String text, boolean isCalculated, HorizontalAlignment align,
			VerticalAlignment valign) {
		this.row = row;
		if (isCalculated) {
			double calclulatedWidth = ((row.getWidth() * width) / 100);
			this.width = (float) calclulatedWidth;
		} else {

```

Why? What I am doing wrong?

Thanks,

Marius

**Comment by Frulenzo** [comment]
Oh, if only want

> I like to arrange two cols/cells with the same width 

Then you just need to say : 

```java
[...]
hRow.createCell(50f, "Test1").setFont(PDType1Font.COURIER);
hRow.createCell(50f, "Test1").setFont(PDType1Font.COURIER);
[...]
```
This first value is % from your row width aka table width and not exact width. I think that was problem :)

We defintely need to documentate that better.

Here is my test case 


```java
@Test
    public void SampleTest8() throws IOException
    {

        //Set margins
        float margin = 10;

        //Initialize Document
        PDDocument doc = new PDDocument();
        PDPage page = addNewPage(doc);

        //Initialize table
        float yTableStart = PDRectangle.A4.getHeight() - 75;
        float tableWidth = PDRectangle.A4.getWidth() - 150;
        BaseTable table = new BaseTable(yTableStart, yTableStart, 75, tableWidth, 75, doc, page, true, true);

        Row<PDPage> hRow = table.createRow(20f);
        hRow.createCell(50, "Test1").setFont(PDType1Font.COURIER);
        hRow.createCell(50, "Test1").setFont(PDType1Font.COURIER);

        table.addHeaderRow(hRow);

        table.draw();
        
        //Save the document
        File file = new File("target/BoxableSample8.pdf");
        System.out.println("Sample file saved at : " + file.getAbsolutePath());
        Files.createParentDirs(file);
        doc.save(file);
        doc.close();
    }
```

Here is my output:
[BoxableSample7.pdf](https://github.com/dhorions/boxable/files/339472/BoxableSample7.pdf)

**Comment by marthie** [comment]
Thanks you, it works!

This is a little bit misleading... :-) I assume an absolute width is expected.

```java
	/**
	 * <p>
	 * Creates a cell with provided width, cell value and default left top
	 * alignment
	 * </p>
	 * 
	 * @param width
	 * @param value
	 * @return
	 */
	public Cell<T> createCell(float width, String value) { ...
```

**Comment by Frulenzo** [comment]
It's not a little bit misleading it's defintely misleading and I will fix that together with other java doc (#67).

This method `createCell(float width, String value)` is calling this Constructor where **isCalculated** boolean is set as true and he has default alignments (top,left).

```java
/**
	 * <p>
	 * Constructs a cell.
	 * </p>
	 *
	 * @param row
	 *            The parent row
	 * @param width
	 *            absolute width in points or in % of table width (depending on
	 *            the parameter {@code isCalculated})
	 * @param text
	 *            The text content of the cell
	 * @param isCalculated
	 *            If {@code true}, the width is interpreted in % to the table
	 *            width
	 * @param align
	 *            The {@link HorizontalAlignment} of the cell content
	 * @param valign
	 *            The {@link VerticalAlignment} of the cell content
	 * @see Cell#Cell(Row, float, String, boolean)
	 */
Cell(Row<T> row, float width, String text, boolean isCalculated, HorizontalAlignment align,
			VerticalAlignment valign)
[...]

```
 Thanks for pointing this out.
